src/portfft/utils.hpp

```diff
@@ -100,22 +100,27 @@ constexpr bool can_cast_safely(const InputType& x) {
  * The function should accept factor size and whether it would be have a BATCH_INTERLEAVED layout or not as an input,
  * and should return a boolean indicating whether or not the factor size can fit in any of the implementation.
  * @param transposed whether or not the factor will be computed in a BATCH_INTERLEAVED format
- * @return
+ * @param encountered_prime_factor A flag to be set if a prime factor which cannot be dispatched to workitem
+ * implementation
+ * @return A factor of the committed size which can be dispatched to either workitem or subgroup implementation
  */
 template <typename F>
-IdxGlobal factorize_input_impl(IdxGlobal factor_size, F&& check_and_select_target_level, bool transposed) {
+IdxGlobal factorize_input_impl(IdxGlobal factor_size, F&& check_and_select_target_level, bool transposed,
+                               bool& encountered_prime_factor) {
```

Comment:
I don't really like this thing where we have return values as mutable inputs. See https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-out-multi